--- files.c	2019-08-06 09:02:06.969340987 -0700
+++ files.c	2019-08-06 09:03:35.346162197 -0700
@@ -160,27 +160,27 @@
 
     int count = 0;
     for (;;) {
-        struct dirent de, *res;
-        if (readdir_r(dir, &de, &res) > 0) {
+        //struct dirent de, *res;
+        if (readdir(dir) != NULL) {
             PLOG_E("Couldn't read the '%s' dir", hfuzz->inputFile);
             closedir(dir);
             return false;
         }
 
-        if (res == NULL && count > 0) {
+        if (count > 0) {
             LOG_I("%zu input files have been added to the list", hfuzz->fileCnt);
             closedir(dir);
             return true;
         }
 
-        if (res == NULL && count == 0) {
+        if (count == 0) {
             LOG_E("Directory '%s' doesn't contain any regular files", hfuzz->inputFile);
             closedir(dir);
             return false;
         }
 
         char path[PATH_MAX];
-        snprintf(path, sizeof(path), "%s/%s", hfuzz->inputFile, res->d_name);
+        snprintf(path, sizeof(path), "%s", hfuzz->inputFile);
         struct stat st;
         if (stat(path, &st) == -1) {
             LOG_W("Couldn't stat() the '%s' file", path);
